sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -111,23 +128,45 @@
             ConfidentialClientApplication.Builder applicationBuilder =
                 ConfidentialClientApplication.builder(clientId, ClientCredentialFactory.createFromSecret(clientSecret))
                     .authority(authorityUrl);
-            if (options.getProxyOptions() != null) {
-                applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
-            }
+
+            // If user supplies the pipeline, then it should override all other properties
+            // as they should directly be set on the pipeline.
             if (options.getHttpPipeline() != null) {
                 applicationBuilder.httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));
+            } else {
+                // If http client is set on the credential, then it should override the proxy options if any configured.
+                if (options.getHttpClient() != null) {
+                    HttpPipeline pipeline = setupPipeline(options.getHttpClient());
+                    applicationBuilder.httpClient(new HttpPipelineAdapter(pipeline));
+                } else if (options.getProxyOptions() != null) {
+                    applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+                }
```

Comment:
should we use the same pipeline that we create for the publicApplicationClient in the constructor so we don't have to create a new pipeline for every call to authenticate through the confidential client?

Reply:
yeah, pipeline construction is now moved to the constructor.